pandas/core/window/rolling.py

```diff
@@ -397,7 +397,7 @@ def _insert_on_column(self, result: "DataFrame", obj: "DataFrame"):
 
         if self.on is not None and not self._on.equals(obj.index):
             name = self._on.name
-            extra_col = Series(self._on, index=obj.index, name=name)
+            extra_col = Series(self._on, index=self.obj.index, name=name).sort_index()
```

Comment:
why the sort?

Reply:
No reason, is a mistake. Thanks for catching it. Deleted